docs/docs/usage/streams/signature-verification.md

```diff
@@ -0,0 +1,57 @@
+---
+sidebar_position: 9
+---
+
+# Signature verification
+Authenticity and integrity of events published on a stream is guaranteed with digital signatures. All messages published to streams are cryptographically signed, which prevents tampering of messages and allows recipients to validate the message's publisher.
+
+:::info Practicalities
+- In the browser, every published message will typically need to be manually signed by metamask. 
+- When viewing a private (encrypted) stream in the browser, Metamask may ask for your signature. This is to verify that you are in the authorised set of the stream's subscribers.
+- Signing is instrumental to the protocol and cannot be turned off. 
+:::
+
+For more in depth analysis, see [Security - Data signing & verification](../../streamr-network/security/signing-and-verification.md).
+
+## Smart contract pub/sub (ERC-1271)
+Beside the typical externally owned account signing and verification, the Streamr protocol also supports smart contract signature verification, i.e. ERC1271 support. This feature allows the use of smart contracts that follow the ERC1271 spec to become signatories on streams in the Streamr Network.
+
+:::info Key points:
+- This feature is enabled on a per stream basis.
+- Ensure the EIP-1271 contract address provided in the stream configuration is valid and compliant with the EIP-1271 specification.
+- Only elliptic curve, secp256k1 cryptography is supported.
+- Smart contract signature validation is periodically [cached](#optimizations-and-caching).
+:::
+
+Every message (data point) on Streamr must be signed with the private key of the publisher and these messages are validated on every hop inside the network by Streamr nodes. With externally owned accounts (EOAs) the Streamr SDK uses `ecrecover` in this process (to verify the public key of the signed message). If the stream has enabled ERC1271 support, the verification will be done through the assigned smart contract with a call to the function `isValidSignature`.
```

Comment:
> these messages are validated on every hop inside the network by Streamr nodes

Actually they are only validated at the the edges of the network, by subscribers. The propagating nodes don't perform validation (because that would increase propagation latency)

Reply:
> If the stream has enabled ERC1271 support, the verification will be done through the assigned smart contract with a call to the function `isValidSignature`.

Harping back to an earlier point, this happens on a "per publishing session" basis when the `erc1271contract` parameter has been set. So it is more a question of how the `streamrClient.publish` method is being called rather than whether an ERC-1271 contract has been given publish permission to the stream.